src/function/probability/lgamma.js

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable no-loss-of-precision */
+
+// References
+// ----------
+// [1] Hare, "Computing the Principal Branch of log-Gamma", Journal of Algorithms, 1997.
+// [2] https://math.stackexchange.com/questions/1338753/how-do-i-calculate-values-for-gamma-function-with-complex-arguments
+
+import { lgammaNumber, lnSqrt2PI } from '../../plain/number/index.js'
+import { factory } from '../../utils/factory.js'
+
+const name = 'lgamma'
+const dependencies = ['Complex', 'typed']
+
+export const createLgamma = /* #__PURE__ */ factory(name, dependencies, ({ Complex, typed }) => {
+  // Stirling series is non-convergent, we need to use the recurrence `lgamma(z) = lgamma(z+1) - log z` to get
+  // sufficient accuracy.
+  //
+  // These two values are copied from Scipy implementation:
+  // https://github.com/scipy/scipy/blob/v1.8.0/scipy/special/_loggamma.pxd#L37
+  const SMALL_RE = 7
+  const SMALL_IM = 7
+
+  /**
+   * The coefficients are B[2*n]/(2*n*(2*n - 1)) where B[2*n] is the (2*n)th Bernoulli number. See (1.1) in [1].
+   *
+   * We can also get these values from the formula (if you cannot access the paper) in [2].
+   *
+   *    1 /     12 = 0.00833333333333333333333333333333
+   *    1 /    360 = 0.00277777777777777777777777777778
+   * ...
+   * 3617 / 133400 = 0.02955065359477124183006535947712
+   */
+  const coeffs = [
+    -2.955065359477124183e-2, 6.4102564102564102564e-3, -1.9175269175269175269e-3, 8.4175084175084175084e-4,
+    -5.952380952380952381e-4, 7.9365079365079365079e-4, -2.7777777777777777778e-3, 8.3333333333333333333e-2
+  ]
+
+  /**
+   * Logarithm of the gamma function for real, positive numbers and complex numbers.
+   *
```

Comment:
It's odd now that the embedded docs, which tend to be brief, have implementation information but the "full" docs do not. Please add that info here as well. (There is a hope that in the not-too-distant future, the embedded docs will also be generated from the jsdoc-style comments as well, so we definitely want all of the info there.)